text/0162-refactoring-support.md

```diff
@@ -0,0 +1,622 @@
+
+    The following resources were moved or renamed:
+
+    ┌───────────────────────────────┬───────────────────────────────┬──────────────────────────┐
+    │ Resource Type                 │ Old Logical ID                │ New Logical ID           │
+    ├───────────────────────────────┼───────────────────────────────┼──────────────────────────┤
+    │ AWS::S3::Bucket               │ MyStack.Bucket5766466B        │ Web.Bucket843D52FF       │
+    ├───────────────────────────────┼───────────────────────────────┼──────────────────────────┤
+    │ AWS::CloudFront::Distribution │ MyStack.DistributionE3BB089E  │ Web.Distribution7142E1F1 │
+    ├───────────────────────────────┼───────────────────────────────┼──────────────────────────┤
+    │ AWS::Lambda::Function         │ MyStack.FunctionA5EA2BD8      │ Service.Function8F0BB69B │
+    └───────────────────────────────┴───────────────────────────────┴──────────────────────────┘
+
+    ? What do you want to do? (Use arrow keys)
+    ❯ Execute the refactor and deploy
+      Deploy without refactoring (will cause resource replacement)
+      Quit
+
+If you choose to refactor and deploy, the CLI will show the progress as the
+refactor is executed:
+
+    Refactoring...
+    Creating stack refactor...
+
+     0/3 | 2:03:17 PM | REFACTOR_IN_PROGRESS | AWS::S3::Bucket               | MyStack.Bucket5766466B       | Web.Bucket843D52FF           
+     0/3 | 2:03:17 PM | REFACTOR_IN_PROGRESS | AWS::CloudFront::Distribution | MyStack.DistributionE3BB089E | Web.Distribution7142E1F1     
+     1/3 | 2:03:18 PM | REFACTOR_COMPLETE    | AWS::S3::Bucket               | MyStack.Bucket5766466B       | Web.Bucket843D52FF           
+     1/3 | 2:03:18 PM | REFACTOR_IN_PROGRESS | AWS::Lambda::Function         | MyStack.FunctionA5EA2BD8     | Service.Function8F0BB69B     
+     2/3 | 2:03:19 PM | REFACTOR_COMPLETE    | AWS::CloudFront::Distribution | MyStack.DistributionE3BB089E | Web.DistributionE3BB089E 
+     3/3 | 2:03:20 PM | REFACTOR_COMPLETE    | AWS::Lambda::Function         | MyStack.FunctionA5EA2BD8     | Service.FunctionA5EA2BD8     
+    
+    ✅  Stack refactor complete
+
+You can configure the refactoring behavior by passing the `--refactoring` flag
+to the CLI, or by configuring this setting in the `cdk.json` file:
+
+```json
+{
+  "app": "...",
+  "refactoring": "EXECUTE_AND_DEPLOY"
+}
+```
+
+Please note that the same CDK application can have multiple stacks for different
+environments. In that case, the CLI will group the stacks by environment and
+perform the refactoring separately in each one. Trying to move resources between
+stacks that belong in different environments will result in an error.
+
+### Rollbacks
+
+After refactoring the stack, the CLI will proceed with the deployment
+(assuming that is your choice). If the deployment fails, and CloudFormation
+rolls it back, the CLI will execute a second refactor, in reverse, to bring the
+resources back to their original locations.
```

Comment:
We will need a flag to not do that. Or maybe the existing `--no-rollback` will do.